pkg/util/password/confirm.go

```diff
@@ -0,0 +1,17 @@
+package password
+
+import (
+	"crypto/subtle"
+
+	"github.com/authgear/authgear-server/pkg/api/apierrors"
+)
+
+var ErrNewPasswordTypo = apierrors.Invalid.WithReason("NewPasswordTypo").New("new password typo")
+
+func ConfirmPassword(newPassword, confirmPassword string) error {
+	typo := subtle.ConstantTimeCompare([]byte(newPassword), []byte(confirmPassword)) == 0
```

Comment:
We don't need constant time comparison here, since both sides are supplied by the same requestor.